### PR TITLE
Miscellaneous documentation fixes

### PR DESCRIPTION
**Pull request by chloelee767**
### What this does
This PR makes the following changes:

Larger changes:
- Update all sequence diagrams to show `ResiRegParser#parseCommand` as a static method
- Moves the `CommandWordEnum` from the `logic.commands` package to the `logic` package, since it is quite tightly coupled to objects from both the `logic.commands` and `logic.parser` packages. This also has the side benefit of making the class diagram for the logic package less messy.

Minor changes:
- Links to addressbook's code (eg. `Logic.java`, `Storage.java`, etc) in the DG have been updated to links to our code instead
- Adds a note about the PlantUML destroy marker to the start of the DG, so that we don't have to keep adding notes to our individual sequence diagrams
- Improves/fixes Javadocs for a couple of the classes I implemented
- Update use cases:
   - remove statistics use case
   - fix use case numbering
   - minor changes to use cases I wrote

### How to test
Read through the updated DG if possible.

Also requesting @jetnew and @Lysire to check that I haven't accidentally butchered your sequence diagrams.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -555,6 +555,9 @@ Priorities: High (must have) - `☆ ☆ ☆`, Medium (nice to have) - `* *`, Low
 | \* \*    | OHS admin                | view all outstanding bills for a student                 | inform the student of his/her due bills.                                                                  |
 | \* \*    | OHS admin                | mark a bill as paid                                      | easily keep track of the remaining amount a student has to pay to OHS.                                    |
 | \* \*    | OHS admin                | view a list of all students with outstanding bills       | remind students of outstanding payments.                                                                  |
+| \* \*    | OHS admin                | add a remark to a bill                                   | record any exceptional details about the bill (e.g. cash-only payment).                                   |
+| \* \*    | advanced user       | have autocompletions for a command                       | quickly and efficiently complete an operation.                                                            |
```
Comment (JingYenLoh) [inline docs/DeveloperGuide.md:561]: Thanks for updating this section! Also, let's have an `F` for this feature :disappointed:

**Review by JingYenLoh — COMMENTED** [review]
There's also a lingering reference to addressbook on line 49, which I believe can be updated to point at our repo. I briefly looked at the diagrams and it seems right, but I'll leave the details to Jet and Kevin.

**Comment by chloelee767** [comment]
> There's also a lingering reference to addressbook on line 49, which I believe can be updated to point at our repo. I briefly looked at the diagrams and it seems right, but I'll leave the details to Jet and Kevin.

Thanks for the catch, misread that line.

**Comment by Lysire** [comment]
Yup, as with what Jing Yen said, my diagram is fine. Also thanks for this PR!

**Review by JingYenLoh — APPROVED** [review]

**Pull request merged**